### PR TITLE
Edu270 4

### PR DESCRIPTION
**Pull request by Maxdzy**
Configure store according to requirements using migration scripts 

Requirements

Store available in 2 different languages via store switcher in the header - Store switcher example  (Just example. The styling does not have to match)

Base currency is EUR, but English store uses GBP 

English store uses scandi/default theme, but the german store uses scandi/german theme. (Both are should be created in the skin folder, but no modification necessary yet)

Product and category pages should not have .html suffix

### CODE REVIEW

File: app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+use Magento\Theme\Model\ResourceModel\Theme\CollectionFactory;
+use Magento\Config\Model\ResourceModel\Config as ConfigResurce;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Store\Model\ResourceModel\Store as StoreResource;
+use Magento\Store\Model\StoreFactory;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Class AddNewCmsPageTask2
```
Comment (Darvex) [inline app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php]: Wrong class name

File: app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+ * @package Edu\CmsCreateSlider\Setup\Patch\Data
```
Comment (Darvex) [inline app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php]: Wrong package

File: app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    /**
+     * {@inheritdoc}
+     * @throws \Exception
+     */
+    public function apply()
+    {
+        $this->moduleDataSetup->startSetup();
+        $themeCollection = $this->collectionFactory->create();
+        $theme = $themeCollection->getThemeByFullPath("frontend/Skin/german");
```
Comment (Darvex) [inline app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php]: Please use same style quotes everywhere (whichever you prefer, but keep it consistent)

File: app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        $themeId = $theme->getId();
+        $websiteid = $this->storeManager->getWebsite()->getId();
+        $groupid = $this->storeManager->getGroup()->getId();
+        $store = $this->storeFactory->create();
+        $store->setName('german');
```
Comment (Darvex) [inline app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php]: You can chain these setters

File: app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        $store->setCode('euro');
+        $store->setWebsiteId($websiteid);
+        $store->setGroupId($groupid);
+        $store->setSortOrder(0);
+        $store->setIsActive(1);
+        $this->storeResource->save($store);
+        $storeId=$store->getId();
+        $configs[] = [
```
Comment (Darvex) [inline app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php]: You can add several values in one assignment command

File: app/design/frontend/Skin/default/etc/view.xml [hunk 6]
```diff
@@ -0,0 +1,315 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<view xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (Darvex) [inline app/design/frontend/Skin/default/etc/view.xml]: Do you really need this file? Seems to me that it should be inherited from parent theme

File: app/design/frontend/Skin/german/etc/view.xml [hunk 7]
```diff
@@ -0,0 +1,315 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<view xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (Darvex) [inline app/design/frontend/Skin/german/etc/view.xml]: Do you need this file? I think it should be inherited from parent theme

**Review by Darvex — CHANGES_REQUESTED** [review]

File: app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php (hunk 4 above)
Comment (Maxdzy) [reply app/code/Edu/CmsCreateStore/Setup/Patch/Data/AddNewCmsStore.php]: what is the best way to write a setters?

**Review by Darvex — APPROVED** [review]

**Pull request merged**